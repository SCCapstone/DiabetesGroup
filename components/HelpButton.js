import React, { Component } from 'react';

import {
    StyleSheet,
    View,
    TouchableHighlight,
    Image,
} from 'react-native';

class HelpButton extends Component {
    render() {
        return (
            <View style={styles.press}>
                <TouchableHighlight
                    onPress={this.props.onPress}>
                    <Image
                        style ={styles.button}
                        source = {require('./help.png')}/>
                </TouchableHighlight>
            </View>
        );
    }
}


const styles = StyleSheet.create({
    press: {
        overflow: 'hidden',
    },
    button: {
        height: 50,
        width: 50,
    },
});

module.exports = HelpButton;