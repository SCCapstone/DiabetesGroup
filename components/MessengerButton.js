import React, { Component } from 'react';

import {
    StyleSheet,
    View,
    TouchableHighlight,
    Image,
} from 'react-native';

class MessengerButton extends Component {
    render() {
        return (
            <View style={styles.press}>
                <TouchableHighlight
                    onPress={this.props.onPress}>
                    <Image
                        source = {require('./MessengerB.png')}/>
                </TouchableHighlight>
            </View>
        );
    }
}


const styles = StyleSheet.create({
    press: {
        overflow: 'hidden',
    },
});

module.exports = MessengerButton;
