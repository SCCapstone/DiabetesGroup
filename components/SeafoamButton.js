import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    View,
    TouchableHighlight,
    TouchableOpacity,
} from 'react-native';

class SeafoamButton extends Component {
    render() {
        return (
            <View style={styles.press}>
                <TouchableOpacity
                    onPress={this.props.onPress}>
                    <Text style={styles.pressText}>{this.props.title}</Text>
                </TouchableOpacity>
            </View>
        );
    }
}


const styles = StyleSheet.create({
    press: {
        backgroundColor: '#112471',
        borderRadius: 12,
        borderColor: '#000000',
        overflow: 'hidden',
        borderWidth: 3,
        paddingLeft: 10,
        paddingRight: 10,
        paddingTop: 10,
        paddingBottom: 10,

    },
    pressText: {
        color: '#FFFFFF',
        fontSize: 16,
        textAlign: 'center',
    },
});

module.exports = SeafoamButton;
