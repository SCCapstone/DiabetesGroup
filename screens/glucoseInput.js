import React, { Component } from 'react';
const SeafoamButton = require('../components/SeafoamButton');
import firebaseApp from './FireBaseApp';
import moment from 'moment';
import {
    Platform,
    StyleSheet,
    Text,
    TextInput,
    View,
    Button,
    AppRegistry,
    TouchableOpacity,
    ScrollView,
    Picker,
} from 'react-native';

export default class glucoseInput extends Component<{}> {

    static navigationOptions = {
        title: 'Glucose Input',
        headerStyle: {backgroundColor: "#FF6127"}
    };

    constructor(props) {
        super(props);
        this.state = {time: '', glucoseLevel: '', readingType:'', notes:''};
    }


    _patientValues() {
        var time = moment().format("MM/DD/YYYY, LT");
        var glucoseLevel = this.state.glucoseLevel;
        var readingType = this.state.readingType;
        var notes = this.state.notes;
        var user = firebaseApp.auth().currentUser;

        if(glucoseLevel < 30 || glucoseLevel > 600)
        {
            alert('Please enter a valid glucose level.');
        }
        else if(readingType === "")
        {
            alert('Please select a valid reading type.');
        }else
        {
            firebaseApp.database().ref('Patients/' + user.uid + '/logs/').push({
                time: time,
                glucoseLevel: glucoseLevel,
                readingType: readingType,
                notes: notes,
            });
            const {navigate} = this.props.navigation;
            navigate('PHome')
        }
    }


    render() {
        return (

                <View style={styles.container}>
                    <View style={styles.stretched}>
                        <Text style={styles.title}>
                            Enter the following fields:
                        </Text>

                        <TextInput style={styles.input} placeholder="Glucose Level"
                                   underlineColorAndroid ={'transparent'}
                                   placeholderTextColor="#CFCFCF"
                                   onChangeText={(text) => this.setState({glucoseLevel: text})}
                                   value={this.state.glucoseLevel}
                        />
                        <Text style={styles.instructions}>
                            mg/dL
                        </Text>

                        <Picker
                            selectedValue={this.state.readingType}
                            onValueChange={(itemValue) => this.setState({readingType: itemValue})}>
                            <Picker.Item label="Reading Type" value="Reading Type"/>
                            <Picker.Item label="Fasting" value="Fasting" />
                            <Picker.Item label="Post-Meal" value="Post-Meal" />
                            <Picker.Item label="Other" value="Other" />
                        </Picker>

                        <TextInput style={styles.input} placeholder="Additional Notes"
                                   underlineColorAndroid ={'transparent'}
                                   placeholderTextColor="#CFCFCF"
                                   onChangeText={(text) => this.setState({notes: text})}
                                   value={this.state.notes}
                        />



                    </View>
                    <SeafoamButton title="Submit"
                    onPress = { () => this._patientValues()}
                    />

                </View>

        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        paddingLeft: 55,
        paddingRight: 55,
        backgroundColor: '#F7F1D2',
    },

    container2: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-around',
    },
    stretched: {
        alignSelf: 'stretch',
    },

    instructions: {
        textAlign: 'center',
        color: '#333333',
        marginBottom: 10,
        marginTop: -20,
    },
    input:{
        fontSize: 16,
        backgroundColor: '#FEFDF5',
        marginBottom: 20,
        borderWidth: 1,

    },
    title: {
        fontSize: 15,
        marginBottom: 10,
        paddingBottom: 5,
        textAlign: 'center',
    }
});

