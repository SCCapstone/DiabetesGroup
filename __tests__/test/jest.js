jest.unmock('../../screens/glucoseInput');

import React from 'react';
import 'react-native';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
chai.use(sinonChai);
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

//global.React = React;
//global.renderer = renderer;
//global.shallow = shallow;
import glucoseInput from '../../screens/glucoseInput';
import MedicationInput from '../../screens/NmedicationInput';
import SeafoamButton from '../../components/SeafoamButton';
import MessengerButton from '../../components/MessengerButton';
import HelpButton from '../../components/HelpButton';
//import GlucoseLogTable from '../../components/GlucoseLogTable';

describe('glucoseInput', () => {
    let component;
    let textInput;
    const defaultState = {text: ''};

    /*test('Medication Input Screen renders correctly', () => {
        const snapshot = renderer.create(<MedicationInput />).toJSON();
        expect(snapshot).toMatchSnapshot();
    });*/

    test('Glucose Input Screen renders correctly', () => {
        const snapshot = renderer.create(<glucoseInput />).toJSON();
        expect(snapshot).toMatchSnapshot();
    });
    test('HelpButton component works', () => {
        expect(typeof HelpButton).toEqual('function');
    });
    test('SeafoamButton component works', () => {
        expect(typeof SeafoamButton).toEqual('function');
    });
    test('MessengerButton component works', () => {
        expect(typeof MessengerButton).toEqual('function');
    });
    test('GlucoseInput component works', () => {
        expect(typeof glucoseInput).toEqual('function');
    });
    /*test('renders the TextInput component', () => {
        const tree = renderer
            .create(<TextInput/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });*/

    /*beforeEach(() => {
        component = shallow(<glucoseInput />);
        textInput = component.find('TextInput');
    });*/

    //placeholder test not working yet

    /*test('has correct placeholder',() => {
        component = shallow(<glucoseInput />);
        textInput = component.find('TextInput');
        const expectedPH = 'Glucose Level';
        /!*const wrapper = mount(<glucoseInput/>);
        expect(wrapper.find)*!/
        expect(textInput.props().placeholder).to.eql(expectedPH);
    })*/


});
