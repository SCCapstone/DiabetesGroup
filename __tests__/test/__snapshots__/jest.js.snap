// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`glucoseInput Glucose Input Screen renders correctly 1`] = `<glucoseInput />`;
